graphbolt/include/graphbolt/sampled_subgraph.h

```diff
@@ -0,0 +1,118 @@
+/**
+ *  Copyright (c) 2023 by Contributors
+ * @file graphbolt/sampled_subgraph.h
+ * @brief Header file of sampled sub graph.
+ */
+
+#ifndef GRAPHBOLT_SAMPLED_SUBGRAPH_H_
+#define GRAPHBOLT_SAMPLED_SUBGRAPH_H_
+
+#include <torch/custom_class.h>
+#include <torch/torch.h>
+
+namespace graphbolt {
+namespace sampling {
+
+/**
+ * @brief Struct representing a sampled subgraph.
+ *
+ * Example usage:
+ *
+ * Suppose the subgraph has 3 nodes and 9 edges.
+ * ```
+ * auto indptr = torch::tensor({0, 2, 3, 4}, {torch::kInt64});
+ * auto indices = torch::tensor({55, 101, 3, 3}, {torch::kInt64});
+ * auto reverse_row_node_ids = torch::tensor({3, 3, 101}, {torch::kInt64});
+ *
+ * SampledSubgraph sampledSubgraph(indptr, indices, reverse_row_node_ids);
+ * ```
+ *
+ * The `reverse_row_node_ids` indicates that nodes `[3, 3, 101]` in the
+ * original graph are mapped to `[0, 1, 2]` in this subgraph, and because
+ * `reverse_column_node_ids` is `Null`, `{55, 101, 3, 3}` in `indices` is just
+ * the original node ids without compaction.
+ *
+ * If `reverse_column_node_ids = torch::tensor({55, 101, 3}, {torch::kInt64})`,
+ * it would indicate a different mapping for the column nodes. Note this is
+ * inconsistent with row, which is legal, as `3` is mapped to `0` and `1` in the
+ * row while `2` in the column.
+ */
+struct SampledSubgraph : torch::CustomClassHolder {
+ public:
```

Comment:
as `struct` is used for class definition, no `public` is needed as it's always public. This public accessible is on purpose?

Reply:
Yes, it is.